pkg/operator/ceph/object/spec.go

```diff
@@ -407,36 +414,52 @@ func configureReadinessProbe(container *v1.Container, healthCheck cephv1.ObjectH
 }
 
 func (c *clusterConfig) defaultLivenessProbe() *v1.Probe {
+	// RGW has internal mechanisms for restarting its processing if it gets stuck. any failures
+	// detected in a liveness probe are likely to either (1) be resolved by the RGW internally or
+	// (2) be a result of connection issues to the Ceph cluster. in the first case, restarting is
+	// unnecessary. in the second case, restarting will only cause more load to the Ceph cluster by
+	// causing RGWs to attempt to re-connect, potentially causing more issues with the Ceph cluster.
+	// forcing a restart of the RGW is more likely to cause issues than solve them, so do not
+	// implement this probe.
+	return nil
+}
+
+func (c *clusterConfig) defaultReadinessProbe() *v1.Probe {
 	return &v1.Probe{
 		ProbeHandler: v1.ProbeHandler{
-			TCPSocket: &v1.TCPSocketAction{
-				Port: c.generateProbePort(),
+			Exec: &v1.ExecAction{
+				Command: []string{
+					"bash", "-c", rgwReadinessProbeScript,
+				},
 			},
 		},
+		TimeoutSeconds:      5,
 		InitialDelaySeconds: 10,
+		// if RGWs aren't responding reliably, remove them from service routing until they are stable
+		PeriodSeconds:    10,
+		FailureThreshold: 3,
+		SuccessThreshold: 3, // don't re-add too soon to "flappy" RGWs from being rout-able
 	}
 }
 
-func (c *clusterConfig) defaultReadinessProbe() *v1.Probe {
+func (c *clusterConfig) defaultStartupProbe() *v1.Probe {
 	return &v1.Probe{
 		ProbeHandler: v1.ProbeHandler{
-			HTTPGet: &v1.HTTPGetAction{
-				Path:   readinessProbePath,
-				Port:   c.generateProbePort(),
-				Scheme: c.generateReadinessProbeScheme(),
+			Exec: &v1.ExecAction{
+				Command: []string{
+					"bash", "-c", rgwStartupProbeScript,
+				},
 			},
 		},
+		TimeoutSeconds:      5,
 		InitialDelaySeconds: 10,
+		// RGW's default init timeout is 300 seconds; give an extra margin before the pod should be
```

Comment:
300s is a long time. Is this to account for resharding or similar maintenance?

Reply:
I'm really not sure. It's documented here: https://docs.ceph.com/en/latest/radosgw/config-ref/#confval-rgw_init_timeout